homeassistant/components/onkyo/media_player.py

```diff
@@ -115,38 +122,21 @@
 SERVICE_SELECT_HDMI_OUTPUT = "onkyo_select_hdmi_output"
 
 
-def determine_zones(receiver):
-    """Determine what zones are available for the receiver."""
-    out = {"zone2": False, "zone3": False}
-    try:
-        _LOGGER.debug("Checking for zone 2 capability")
-        receiver.raw("ZPW")
-        out["zone2"] = True
-    except ValueError as error:
-        if str(error) != TIMEOUT_MESSAGE:
-            raise error
-        _LOGGER.debug("Zone 2 timed out, assuming no functionality")
-    try:
-        _LOGGER.debug("Checking for zone 3 capability")
-        receiver.raw("PW3")
-        out["zone3"] = True
-    except ValueError as error:
-        if str(error) != TIMEOUT_MESSAGE:
-            raise error
-        _LOGGER.debug("Zone 3 timed out, assuming no functionality")
-
-    return out
-
-
-def setup_platform(hass, config, add_entities, discovery_info=None):
-    """Set up the Onkyo platform."""
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Set up our socket to the AVR."""
+
     host = config.get(CONF_HOST)
-    hosts = []
+    port = config.get(CONF_PORT)
+    name = config.get(CONF_NAME) or "Onkyo Reciever"
+    max_volume = config.get(CONF_MAX_VOLUME)
+    receiver_max_volume = config.get(CONF_RECEIVER_MAX_VOLUME)
+    zones = config.get(CONF_ZONES)
```

Comment:
```suggestion
    zones = config[CONF_ZONES]
```
This has a default so it should always have a value